scipy/stats/arb_dist.py

```diff
@@ -0,0 +1,269 @@
+    xtol : float, optional
+        The tolerance for fixed point calculation for generic ppf.
+    interpolation : int, optional
+        If `y` represents a non-histogrammed sampled PDF (i.e.
+        ``y.size == x.size``), then `interpolation` specifies the order of an
+        interpolating spline between each of the sampled points. The default of
+        1 represents linear interpolation, 2 is quadratic interpolation, etc.
+        This value should be an integer in the range [1-5].
+
+    Examples
+    --------
+    Let us consider a quadratic distribution specified by a function:
+    >>> from scipy.stats import rv_arbitrary
+    >>> quadratic = lambda x: -x**2 + 1
+    >>> arb = rv_arbitrary(quadratic, [-1, 1])
+    >>> arb.pdf([-1, 0, 0.5, 1])
+    array([0., 0.75, 0.5625, 0.])
+    >>> arb.cdf([-1, 0, 0.5, 1])
+    array([0., 0.5, 0.84375, 1.])
+
+    Define the same distribution using samples. Using
+    ``interpolation=2`` means that quadratic interpolation is used, and the
+    following should be exactly the same as the previous example
+    >>> arb2 = rv_arbitrary([0., 0.75, 0], [-1, 0, 1], interpolation=2)
+    >>> arb2.pdf([-1, 0, 0.5, 1])
+    array([0., 0.75, 0.5625, 0.])
+    >>> arb2.cdf([-1, 0, 0.5, 1])
+    array([0., 0.5, 0.84375, 1.])
+
+    A histogram example
+    >>> x = [0, 2, 3]
+    >>> y = [5, 4]
+    >>> arb3 = rv_arbitrary(y, x)
+    >>> arb3.cdf([0, 1, 2, 3])
+    array([0. , 0.35714286, 0.71428571, 1.])
+    """
+    def __init__(self, y, x, xtol=1e-14, seed=None, interpolation=1):
+        self._y = y
+        self._x = np.asfarray(x)
+
+        # data can be a histogram or a sampled function.
+        szx = np.size(self._x)
+        if szx < 2:
+            raise ValueError("At least two points in `x` are required to"
+                             "define the support for arbitrary distributions")
+
+        if callable(self._y):
+            # TODO: doesn't make sense for szx < 2, or for less than
+            # that required for the interpolating spline
+            self._init_func()
+        else:
+            self._y = np.array(y, float)
+
+            if np.any(self._y < 0):
+                raise ValueError("y specifies a probability distribution, all"
+                                 " values must be >= 0")
+
+            # data can be a histogram or a sampled function.
+            szy = np.size(self._y)
+
+            if szx == szy:
+                self._init_samp(interpolation)
+            elif szx == szy + 1:
+                self._init_hist()
+            else:
+                raise ValueError("The x.size must equal y.size for a"
+                                 " sampled function, or y.size + 1 for a"
+                                 " sampled histogram")
+
+        super(rv_arbitrary, self).__init__(a=np.min(self._x),
+                                           b=np.max(self._x),
+                                           xtol=xtol, seed=seed)
```

Comment:
This will need to avoid the messing with the docstring done at the end of ``rv_continuous.__init__``.